junebug/stores.py

```diff
@@ -1,6 +1,17 @@
 from twisted.internet.defer import inlineCallbacks, returnValue
+from twisted.web import http
+
 from vumi.message import TransportUserMessage
 
+from junebug.error import JunebugError
+
+
+class MessageNotFound(JunebugError):
+    '''Raised when a message is not found.'''
+    name = 'MessageNotFound'
+    description = 'message not found'
+    code = http.NOT_FOUND
```

Comment:
Not sure if this is the right code to use. I think 404 responses are usually for telling the client that the requested entity was not found, not that something needed to fulfill the request was not found.